src/dash-table/syntax-tree/lexeme/relational.ts

```diff
@@ -73,6 +83,24 @@ export const greaterThan: IUnboundedLexeme = R.merge({
     regexp: /^(>|gt)/i
 }, LEXEME_BASE);
 
+const DATE_OPTIONS: IDateValidation = {
+    allow_YY: true
+};
+
+export const likeDate: IUnboundedLexeme = R.merge({
+    evaluate: relationalEvaluator(([op, exp]) => {
+        const normalizedOp = normalizeDate(op, DATE_OPTIONS);
+        const normalizedExp = normalizeDate(exp, DATE_OPTIONS);
+
+        return !R.isNil(normalizedOp) &&
+            !R.isNil(normalizedExp) &&
+            // IE11 does not support `startsWith`
+            normalizedOp.indexOf(normalizedExp) === 0;
```

Comment:
if we're worried about perf, possibly faster to avoid `indexOf`:
```js
(
    normalizedOp.length === normalizedExp.length ?
    normalizedOp : normalizedOp.substr(0, normalizedExp.length)
) === normalizedExp
```